src/main/java/com/xiaomi/infra/pegasus/rpc/interceptor/SecurityReplicaSessionInterceptor.java

```diff
@@ -41,4 +42,13 @@ public void onConnected(ReplicaSession session) {
     Negotiation negotiation = new Negotiation(session, subject, serviceName, serviceFqdn);
     negotiation.start();
   }
+
+  public boolean onSendMessage(ReplicaSession session, final ReplicaSession.RequestEntry entry) {
+    // tryPendRequest returns false means that the negotiation is succeed now
+    return isNegotiationRequest(entry) || !session.tryPendRequest(entry);
+  }
+
+  private boolean isNegotiationRequest(final ReplicaSession.RequestEntry entry) {
+    return entry.op.getClass().equals(negotiation_operator.class);
```

Comment:
using `instanceof` to judge object type, see https://github.com/XiaoMi/pegasus-java-client/blob/3615cfd978797334cb7beec4b624473642698a63/src/main/java/com/xiaomi/infra/pegasus/rpc/interceptor/CompressionInterceptor.java#L37